oscar_odin/mappings/order.py

```diff
@@ -33,17 +39,82 @@ class SurchargeToResource(OscarBaseMapping):
     to_obj = resources.order.Surcharge
 
 
+class DiscountLineToResource(OscarBaseMapping):
+    """Mapping from an order discount line model to a resource"""
+
+    from_obj = OrderLineDiscountModel
+    to_obj = resources.order.DiscountLine
+
+    @odin.assign_field
+    def line(self):
+        """map line object as resource"""
+        return LineToResource.apply(self.source.line)
+
+
 class DiscountToResource(OscarBaseMapping):
     """Mapping from an order discount model to a resource."""
 
     from_obj = OrderDiscountModel
     to_obj = resources.order.Discount
 
     @odin.map_field
-    def category(self, value: str) -> resources.order.DiscountCategory:
+    def category(self, value: str):
         """Map category."""
         return resources.order.DiscountCategory(value)
 
+    @odin.map_field(from_field="category")
+    def is_basket_discount(self, category) -> bool:
+        """map is basket discount function"""
+        return category == "Basket"
+
+    @odin.map_field(from_field="category")
+    def is_shipping_discount(self, category) -> bool:
+        """map is shipping discount function"""
+        return category == "Shipping"
+
+    @odin.map_field(from_field="category")
+    def is_post_order_action(self, category) -> bool:
+        """map is post order action function"""
+        return category == "Deferred"
+
+    @odin.assign_field
+    def description(self) -> str:
+        """map description function"""
+        return self.source.description()
+
+    @odin.assign_field(to_list=True)
+    def discount_lines(self):
+        """map discount lines"""
+        return map_queryset(
+            DiscountLineToResource, self.source.discount_lines, context=self.context
+        )
+
+    @odin.assign_field(to_list=True)
+    def discount_lines_per_tax_code(self):
```

Comment:
This definitely needs changing, first loop trough the discounts and figure out the tax codes per discount. 

Also please add a test that checks the result when no tax codes are present, since this is possible in oscar.